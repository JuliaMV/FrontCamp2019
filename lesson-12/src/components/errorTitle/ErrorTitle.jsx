import React from 'react';
import PropTypes from 'prop-types';

import css from './ErrorTitle.module.css';

const ErrorTitle = ({ title }) => (
  <p className={css.title}>{ title }</p>
);

ErrorTitle.propTypes = {
  title: PropTypes.string.isRequired,
};


export default ErrorTitle;
