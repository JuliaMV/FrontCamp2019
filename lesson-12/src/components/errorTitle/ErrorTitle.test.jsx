import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import ErrorTitle from './ErrorTitle';

Enzyme.configure({ adapter: new Adapter() });

const title = 'Something went wrong...'

describe('ErrorTitle component', () => {
  test(`should render with title "${title}"`, () => {
    const props = { title };
    const component = shallow(<ErrorTitle title={props.title} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
