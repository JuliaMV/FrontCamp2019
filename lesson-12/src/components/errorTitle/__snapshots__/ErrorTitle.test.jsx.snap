// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorTitle component should render with title "Something went wrong..." 1`] = `
<p>
  Something went wrong...
</p>
`;
